### ISSUE TITLE
Broken autocomplete with desc ordering

### ISSUE THREAD

**Issue by clutcher** [sehmaschine/django-grappelli#701]
Using descending ordering in model breaks autocomplete.

In my case, model:
```python
class City(AbstractCity):
    @staticmethod
    def autocomplete_search_fields():
        return ("id__iexact", "name_ascii__icontains",)

    class Meta:
        ordering = ['-population']
```

causes "{Field error} Cannot resolve keyword '-population'" in views/related.py:165
```python
distinct_columns = list(self.model._meta.ordering) + [self.model._meta.pk.column]
return qs.order_by(*distinct_columns).distinct(*distinct_columns)
```

Using just ```return qs.order_by(*distinct_columns).distinct()``` solves the problem.

Package versions:
Django==1.8.7
django-grappelli==2.7.2

**Comment by ninjaa** [comment]
Same issue

**Comment by ninjaa** [comment]
Can verify @clutcher's fix seems to work as well ... the following is my monkey patch

```python
# Monkey patch grappelli autocomplete to fix bug
# https://github.com/sehmaschine/django-grappelli/issues/701
from django.db import connection
from grappelli.views.related import AutocompleteLookup 

def grp_autocomplete_get_queryset(self):
    qs = super(AutocompleteLookup, self).get_queryset()
    qs = self.get_filtered_queryset(qs)
    qs = self.get_searched_queryset(qs)
    if connection.vendor == 'postgresql':
        distinct_columns = list(self.model._meta.ordering) + [self.model._meta.pk.column]
        return qs.order_by(*distinct_columns).distinct()
    else:
        return qs.distinct()

setattr(AutocompleteLookup, 'get_queryset', grp_autocomplete_get_queryset)
```

**Comment by sehmaschine** [comment]
@clutcher @ninjaa I see what you mean. But with the proposed "fix", we do have another issue - see https://github.com/sehmaschine/django-grappelli/pull/646

**Comment by sehmaschine** [comment]
should be solved, see #685
